src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2261,29 +2261,22 @@ private void WriteToStandardInput(Process p)
             writer.Dispose();
         }
 #else
-        private SafeFileHandle GetSafeFileHandleForRedirection(string RedirectionPath, uint dwCreationDisposition)
-        {
-            System.IntPtr hFileHandle = System.IntPtr.Zero;
-            ProcessNativeMethods.SECURITY_ATTRIBUTES lpSecurityAttributes = new();
-
-            hFileHandle = ProcessNativeMethods.CreateFileW(RedirectionPath,
-                ProcessNativeMethods.GENERIC_READ | ProcessNativeMethods.GENERIC_WRITE,
-                ProcessNativeMethods.FILE_SHARE_WRITE | ProcessNativeMethods.FILE_SHARE_READ,
-                lpSecurityAttributes,
-                dwCreationDisposition,
-                ProcessNativeMethods.FILE_ATTRIBUTE_NORMAL,
-                System.IntPtr.Zero);
-            if (hFileHandle == System.IntPtr.Zero)
```

Comment:
There's an actual bug you are fixing here. `CreateFileW` will return `INVALID_HANDLE_VALUE` on an error which is `(IntPtr)(-1)` and not `IntPtr.Zero` so the current code isn't validating whether the handle was opened or not. So now with your change it fixes this bug as it will raise the Win32Exception on an invalid path which it did not before.

You can even see it not fail today by doing

```powershell
Start-Process pwsh -ArgumentList '-c echo "hi"' -RedirectStandardOutput C:\missing\path\test.txt -Wait
```